### PR TITLE
Implementing Hamiltonian Neural network ( HNN)  tutorial

### PR DESCRIPTION
**Pull request by a-b-h-a-y-s-h-i-n-d-e**
## Description

This PR implements tutorial on Hamiltonian Neural Networks along with required changes in docs.

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ x ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ x ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: examples/tutorials/hnn_tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,536 @@
+{
```
Comment (JoseAntonioSiguenza) [inline examples/tutorials/hnn_tutorial.ipynb]: <p>Please start with a brief introduction about deepchem and HNN model. HNNs are well explained, but talk about deepchem.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4497/discussion/#comment-2226142011'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (JoseAntonioSiguenza) [inline examples/tutorials/hnn_tutorial.ipynb]: <p>Please add a line or two about the symplectic gradient and include references for figures and literature.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4497/discussion/#comment-2226142020'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (JoseAntonioSiguenza) [inline examples/tutorials/hnn_tutorial.ipynb]: <p>Please try to follow deepchem tutorial structure (setup, intro, etc). For example, change this to conda install and  include a colab link under "Setup" section.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4497/discussion/#comment-2226142024'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (JoseAntonioSiguenza) [inline examples/tutorials/hnn_tutorial.ipynb]: <p>Please add "Congratulations! Time to join the Community!" "Star DeepChem on&nbsp;<a href="https://github.com/deepchem/deepchem" target="_blank">GitHub</a>" and "Join the DeepChem Gitter".</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4497/discussion/#comment-2226142030'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (a-b-h-a-y-s-h-i-n-d-e) [reply examples/tutorials/hnn_tutorial.ipynb]: done

Comment (a-b-h-a-y-s-h-i-n-d-e) [reply examples/tutorials/hnn_tutorial.ipynb]: <p>done</p>

File: examples/tutorials/hnn_tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
```
Comment (JoseAntonioSiguenza) [inline examples/tutorials/hnn_tutorial.ipynb]: <p>Please introduce Deepchem before talking about HNNs, and when talking about HNNs please add their importance (reduce errors that accumulate over iterations and may lead to inaccurate results in physical systems like mass-springs, etc).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4497/discussion/#comment-2228848122'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (JoseAntonioSiguenza) [inline examples/tutorials/hnn_tutorial.ipynb]: <p>I believe it would be better to have this explanation section after the code execution. Please update this: "This methods defines/implements..." to "This method applies..." or use another word to improve clarity. I see an unexpected <code>$</code> symbol in Hamiltonian equation, please fix. Please change "to get correct "Ground Truth" to "to get "ground truth". Also, please change capitalized "Symplectic" to "symplectic".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4497/discussion/#comment-2228848129'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (JoseAntonioSiguenza) [inline examples/tutorials/hnn_tutorial.ipynb]: <p>Please merge this code block as one (dataset, model and train). Also, please add a line or two describing this steps.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4497/discussion/#comment-2228848135'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (JoseAntonioSiguenza) [inline examples/tutorials/hnn_tutorial.ipynb]: <p>Please add a brief description of expected model outputs</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4497/discussion/#comment-2228848139'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (a-b-h-a-y-s-h-i-n-d-e) [reply examples/tutorials/hnn_tutorial.ipynb]: <p>done</p>

Comment (a-b-h-a-y-s-h-i-n-d-e) [reply examples/tutorials/hnn_tutorial.ipynb]: <p>done</p>

Comment (a-b-h-a-y-s-h-i-n-d-e) [reply examples/tutorials/hnn_tutorial.ipynb]: <p>done</p>

File: examples/tutorials/hnn_tutorial.ipynb (hunk 1 above)
Comment (a-b-h-a-y-s-h-i-n-d-e) [reply examples/tutorials/hnn_tutorial.ipynb]: <p>done</p>

Comment (a-b-h-a-y-s-h-i-n-d-e) [reply examples/tutorials/hnn_tutorial.ipynb]: <p>done</p>

**Comment by JoseAntonioSiguenza** [comment]
LGTM
